src/dist/http.rs

```diff
@@ -11,556 +11,609 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+#![allow(unused)]
+
+#[cfg(feature = "dist-client")]
+pub use self::client::Client;
+#[cfg(feature = "dist-server")]
+pub use self::server::Scheduler;
+#[cfg(feature = "dist-server")]
+pub use self::server::Server;
+
+//#[allow(unused)]
+mod common {
+    use bincode;
+    use futures::{Future, Stream};
+    use reqwest;
+    use serde;
+    use std::net::{IpAddr, SocketAddr};
+    use dist::{JobId, CompileCommand};
+
+    use errors::*;
+
+    const SCHEDULER_PORT: u16 = 10500;
+    const SERVER_PORT: u16 = 10501;
+
+    // TODO: move this into the config module
+    pub struct Cfg;
+
+    impl Cfg {
+        pub fn scheduler_listen_addr() -> SocketAddr {
+            let ip_addr = "0.0.0.0".parse().unwrap();
```

Comment:
I tend to prefer to write out the `Ipv4Addr` longhand, like: https://github.com/mozilla/sccache/blob/1d899441ffa9715132bd6e67dcc5012ce1a70dea/src/server.rs#L198